packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -1086,19 +1089,39 @@ export function mountEventListener(listener: ReactDOMListener): void {
     const {target} = listener;
     if (target === window) {
       // TODO (useEvent)
-    } else {
+    } else if (isDOMDocument(target)) {
+      // TODO (useEvent)
+    } else if (isDOMElement(target)) {
       attachElementListener(listener);
+    } else {
+      // The user should never get to here or we missed something in
+      // validateEventListenerTarget below.
+      invariant(
+        false,
+        'A useEvent listener target instance was not handled. This error is ' +
+          'likely caused by a bug in React. Please file an issue.',
+      );
     }
   }
 }
 
 export function unmountEventListener(listener: ReactDOMListener): void {
   if (enableUseEventAPI) {
     const {target} = listener;
-    if (target === window) {
+    if (isDOMElement(target)) {
```

Comment:
You have `if (isDOMElement(target)) {` and `else if (isDOMElement(target)) {`

I think you meant for this one to be `target === window`.